test/integration/addon/addon_manager_upgrade_test.go

```diff
@@ -793,6 +841,12 @@ var _ = ginkgo.Describe("Addon upgrade", func() {
 					Reason:  addonapiv1alpha1.ProgressingReasonCompleted,
 					Message: "completed with no errors.",
 				})
+				assertManagedClusterAddOnConditions(testAddOnConfigsImpl.name, clusterNames[i], metav1.Condition{
+					Type:    addonapiv1alpha1.ManagedClusterAddOnConditionConfigured,
+					Status:  metav1.ConditionTrue,
+					Reason:  "ConfigurationsConfigured",
+					Message: "Configurations configured",
+				})
```

Comment:
we need an integration test for the case that only when condition is set, the addon-template will start render.

Reply:
will do.